tests/backup/backup_kdmp_test.go

```diff
@@ -1400,3 +1401,158 @@ var _ = Describe("{ExcludeInvalidDirectoryFileBackup}", func() {
 
 	})
 })
+
+// This TC takes 30 generic backups, deletes intermittent backups and validates the restores
+var _ = Describe("{IssueGenericBackupsAndRestoreInterleavedCopies}", func() {
+	var (
+		scheduledAppContexts []*scheduler.Context
+		sourceClusterUid     string
+		backupLocationMap    map[string]string
+		cloudAccountName     string
+		bkpLocationName      string
+		cloudCredUID         string
+		backupLocationUID    string
+		currentBackupName    string
+		bkpNamespaces        []string
+		providers            []string
+		backupNameList       []string
+		backupListForRestore []string
+		restoreNames         []string
+		genericBackupCount   = 30
+		backupDeleteCount    = 15
+		backupRestoreCount   = 4
+	)
+
+	JustBeforeEach(func() {
+		StartPxBackupTorpedoTest("IssueGenericBackupsAndRestoreInterleavedCopies", "To verify the restores for generic backups when intermittent backups are deleted", nil, 58049, Sabrarhussaini, Q3FY24)
+		backupLocationMap = make(map[string]string)
+		log.InfoD("scheduling applications")
+		scheduledAppContexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts := ScheduleApplications(taskName)
+			for _, appCtx := range appContexts {
+				appCtx.ReadinessTimeout = appReadinessTimeout
+				scheduledAppContexts = append(scheduledAppContexts, appCtx)
+				namespace := GetAppNamespace(appCtx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+		providers = getProviders()
+	})
+
+	It("To verify the restores for generic backups when intermittent backups are deleted", func() {
+		Step("Validate applications", func() {
+			log.InfoD("Validating applications")
+			ValidateApplications(scheduledAppContexts)
+		})
+
+		Step("Validate creation of cloud credentials and backup location", func() {
+			log.InfoD("Validate creation of cloud credentials and backup location")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, provider := range providers {
+				cloudCredUID = uuid.New()
+				cloudAccountName = fmt.Sprintf("%s-%s-%v", "cred", provider, RandomString(4))
```

Comment:
create constant CredName here instead of "cred"

Reply:
@sn-px I'm sorry, I couldn't comprehend the comment. Can you please let me know what exact value should be considered? now the format for the cloudAccountName will be "cred-aws-xxxx", did you mean it should be instead "aws-CredName-xxxx  "

Reply:
@sn-px Addressed the comment.